File: solutions/php/prime-factors/1/PrimeFactors.php
<?php
declare(strict_types=1);

function factors(int $number): array
{
    if ($number < 2) {
        return [];
    }
    
    $factors = [];
    $divisor = 2;
    
    while ($divisor * $divisor <= $number) {
        while ($number % $divisor == 0) {
            $factors[] = $divisor;
            $number = intdiv($number, $divisor);
        }
        $divisor++;
    }
    
    if ($number > 1) {
        $factors[] = $number;
    }
    
    return $factors;
}
?>

File: solutions/ruby/prime-factors/1/prime_factors.rb
class PrimeFactors
  def self.of(number)
    factors = []
    divisor = 2

    # El ciclo continúa hasta que el número se reduzca a 1
    while number > 1
      if number % divisor == 0
        factors << divisor
        number /= divisor
      else
        # Si no es divisible de forma exacta, pasamos al siguiente número
        divisor += 1
      end
    end

    factors
  end
end